addons/Hierarchical_Table_Of_Contents/test/KeyboardNavigationTests.js

```diff
@@ -0,0 +1,244 @@
+                </tr>
+            `);
+
+            return $tr;
+    },
+
+    generateTableRowWithId: function (id) {
+            let $tr = $(`
+                <tr class="treegrid-${id}">
+                    <td>
+                        <div class="text-wrapper">
+                            Page
+                        </div>
+                    </td>
+                </tr>
+            `);
+
+            return $tr;
+    },
+
+    getTreeGrid: function (id) {
+        return this.presenter.$view.find(`.treegrid-${id}`);
+    },
+
+    setCollapsedChapterAsCurrentElement: function () {
+        KeyboardController.prototype.switchElement.call(this.presenter.keyboardControllerObject, 1);
+    },
+
+    setLastKeyNavElementAsCurrentElement: function () {
+        KeyboardController.prototype.switchElement.call(this.presenter.keyboardControllerObject, 1);
+    },
+
+    setTableRowAfterCollapsedChapterAsCurrentElement: function () {
+        KeyboardController.prototype.switchElement.call(this.presenter.keyboardControllerObject, 3);
+    },
+
+    activateTTSWithoutReading: function() {
+        this.presenter.setWCAGStatus(true);
+        this.presenter.isFirstEnter = false;
+        this.presenter.keyboardControllerObject.keyboardNavigationActive = true;
+        this.presenter.keyboardControllerObject.setElements(this.presenter.getElementsForTTS());
+    },
+
+    'test given view when entering for the first time by keyboard navigation should mark first hyper link' : function() {
+        activateEnterEvent(this.presenter);
+
+        assertTrue(hasKeyboardNavigationActiveElementClass(this.presenter.$view.find(".treegrid-1")));
+    },
+
+    'test given view with keyNav when next element is collapsed then skip to next one' : function() {
+        activateEnterEvent(this.presenter);
+
+        this.setCollapsedChapterAsCurrentElement();
+
+        activateTabEvent(this.presenter);
+
+        assertTrue(hasKeyboardNavigationActiveElementClass(this.presenter.$view.find(".treegrid-4")));
+    },
+
+    'test given view with keyNav when previous element is collapsed then skip to next one' : function() {
+        activateEnterEvent(this.presenter);
+
+        this.setTableRowAfterCollapsedChapterAsCurrentElement();
+
+        activateShiftTabEvent(this.presenter);
+
+        assertTrue(hasKeyboardNavigationActiveElementClass(this.presenter.$view.find(".treegrid-2")));
+    },
+
+    'test given current element expanded when switching to next then should get first child' : function() {
+        activateEnterEvent(this.presenter);
+
+        this.setCollapsedChapterAsCurrentElement();
+
+        const nestedTreeGrid = this.getTreeGrid(3);
+        setVisible(nestedTreeGrid);
+
+        activateTabEvent(this.presenter);
+
+        assertTrue(hasKeyboardNavigationActiveElementClass(nestedTreeGrid));
+    },
+
+    'test given first enter when TTS active then get title' : function() {
+        this.activateTTSWithoutReading(this.presenter);
```

Comment:
To nie jest w pełni poprawnie zrobiony testy.
activateTTSWithoutReading ustawia keyboardNavigationActive na `true`.

Celem testu na pierwsze wejście jest sprawdzenie wejścia w addon (za pomocą klawiszu enter), gdy nawigacja klawiaturowa nie jest aktywna (np. keyboardNavigationActive jest na false, itd.).
To ze względu na funckjonalność `KeyboardController.prototype.enter`
Czyli w tym przypadku powinno być tylko ;)
```suggestion
        this.presenter.setWCAGStatus(true);
```